src/utils/series.ts

```diff
@@ -1,85 +1,103 @@
-
-  return {
-    ...playlist,
-    playlist: playlist.playlist.filter(({ seasonNumber }) => seasonNumber === filter),
-  };
+/**
+ * Get an array of options for a season filter
+ */
+export const getFiltersFromSeries = (series: Series | undefined): string[] => {
+  return series && 'seasons' in series ? series.seasons.map((season) => String(season.season_number)) : [];
 };
 
-export const getSeriesEpisodes = (series: Series) => {
-  if ('seasons' in series) {
-    return series.seasons.flatMap((season: Season) => season.episodes);
-  }
-
-  return series.episodes;
+/**
+ * Get a playlist with episodes based on the selected filter
+ */
+export const filterSeries = (playlist: Playlist, episodes: EpisodesWithPagination[] | undefined): Playlist => {
+  // Get a flattened list of episodes and return it as part of the playlist
+  return { ...playlist, playlist: episodes?.flatMap((e) => e.episodes) || [] };
 };
 
-/** Next episode to show after video is complete */
-export const getNextEpisode = (series: Series, media: PlaylistItem): string => {
-  // To handle array elements
-  const episodes = getSeriesEpisodes(series);
+/**
+ * Retrieve a first episode of the next season
+ */
+const getNextSeasonEpisode = async (seasonNumber: number, series: Series | undefined) => {
+  const seasons = series?.seasons;
+
+  // The seasons is the last one in case there is only one season available or it is the same as the last season in the seasons array
+  const isLastSeason = seasons?.length === 1 || seasons?.[seasons?.length - 1]?.season_number === seasonNumber;
 
-  const episodeIndex = episodes?.findIndex((el) => el.media_id === media.mediaid);
+  if (isLastSeason) {
+    return;
+  } else {
+    // Need to know the order of seasons to choose the next season number
+    const hasAscendingOrder = (Number(seasons?.[1]?.season_number) || 0) > (Number(seasons?.[0]?.season_number) || 0);
+    const nextSeasonNumber = hasAscendingOrder ? seasonNumber + 1 : seasonNumber - 1;
 
-  // If we have the last episode => we leave the same video
-  if (episodeIndex === episodes.length - 1) {
-    return media.mediaid;
+    return (await getSeasonWithEpisodes(series?.series_id, nextSeasonNumber, 0, 1))?.episodes?.[0];
   }
+};
 
-  return episodes[episodeIndex + 1]?.media_id;
+/**
+ * Get a next episode of the selected season / episodes list
+ */
+const getNextEpisode = async (seasonNumber: number, seriesId: string, pageWithEpisode: number) => {
+  if (seasonNumber) {
+    return (await getSeasonWithEpisodes(seriesId, seasonNumber, pageWithEpisode, 1))?.episodes?.[0];
+  } else {
+    return (await getEpisodes(seriesId, pageWithEpisode, 1))?.episodes?.[0];
+  }
 };
 
-export const getNextItem = (item: PlaylistItem | undefined, series: Series | undefined, seriesPlaylist: Playlist | undefined): PlaylistItem | undefined => {
-  if (!item || !seriesPlaylist) return;
+const getNewFlowNextEpisode = async (series: Series | undefined, episodeMetadata: EpisodeMetadata | undefined) => {
```

Comment:
The easiest way to do this logic would be to keep a flat list of episodes sorted as delivered by the BE. This probably can and should leverage caching. Then to get the next, we just find the index of the current by episode ID and then get the item at index+1.

Reply:
So I've been doing a bit of brainstorming to try to figure out how this would work, but I'm not sure if it's practical. Ideally we would just grab episodes once each and be able to jump around with the paging and have it eventually all work out, but that would require a bunch of FE logic to handle paging better. For now we can leave this mostly as is, but will suggest replacing it with the a single call to `/episodes?page_length=1&after=<current ID>`